concent_api/gatekeeper/views.py

```diff
@@ -40,102 +42,109 @@ def upload(request):
     # Decode and check if request header contains a golem message:
     if 'HTTP_AUTHORIZATION' not in request.META:
-        return gatekeeper_access_denied_response("Missing 'Authorization' header.")
+        return gatekeeper_access_denied_response("Missing 'Authorization' header.", path_to_file)
 
     authorization_scheme_and_token = request.META['HTTP_AUTHORIZATION'].split(" ", 1)
     assert len(authorization_scheme_and_token) in [1, 2]
 
     if len(authorization_scheme_and_token) == 1:
-        return gatekeeper_access_denied_response("Missing token in the 'Authorization' header.")
+        return gatekeeper_access_denied_response("Missing token in the 'Authorization' header.", path_to_file)
 
     (scheme, token) = authorization_scheme_and_token
 
     if scheme != 'Golem':
-        return gatekeeper_access_denied_response("Unrecognized scheme in the 'Authorization' header.")
+        return gatekeeper_access_denied_response("Unrecognized scheme in the 'Authorization' header.", path_to_file)
 
     try:
         decoded_auth_header_content = b64decode(token, validate = True)
     except binascii.Error:
-        return gatekeeper_access_denied_response("Unable to decode token in the 'Authorization' header.")
+        return gatekeeper_access_denied_response("Unable to decode token in the 'Authorization' header.", path_to_file)
 
     try:
         loaded_golem_message = load(decoded_auth_header_content, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY, check_time = False)
     # FIXME: We want to catch only exceptions caused by malformed messages but golem-messages does not have specialized
     # exception classes for that. It simply raises AttributeError.
     except AttributeError:
-        return gatekeeper_access_denied_response("Token in the 'Authorization' header is not a valid Golem message.")
+        return gatekeeper_access_denied_response("Token in the 'Authorization' header is not a valid Golem message.", path_to_file)
 
     if loaded_golem_message is None:
-        return gatekeeper_access_denied_response('Undefined golem message.')
+        return gatekeeper_access_denied_response('Undefined golem message.', path_to_file)
     assert isinstance(loaded_golem_message, Message)
+    logger.debug(
+        "Message operation: {}, subtask_id: {}, path to file: '{}'".format(
+            loaded_golem_message.operation,
+            loaded_golem_message.subtask_id,
+            path_to_file
+        )
+    )
 
     # Check if request header contains Concent-Client-Public-Key:
     if 'HTTP_CONCENT_CLIENT_PUBLIC_KEY' not in request.META:
-        return gatekeeper_access_denied_response('Missing Concent-Client-Public-Key header.')
+        return gatekeeper_access_denied_response('Missing Concent-Client-Public-Key header.', path_to_file, loaded_golem_message.subtask_id)
+    client_public_key = request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY']
+    logger.debug("Concent client public key: {}".format(client_public_key))
```

Comment:
The public key was supposed to be included in the messages about upload/download request, not separately. We want to be able to pull it out of message in Stackdriver and filter by it.